airflow/providers/apache/hive/hooks/hive.py

```diff
@@ -78,9 +78,13 @@ class HiveCliHook(BaseHook):
     :type  mapred_job_name: str
     """
 
+    conn_name_attr = 'hive_cli_conn_id'
+    default_conn_name = 'hive_cli_default'
+    conn_type = 'hive_cli'
+
     def __init__(
         self,
-        hive_cli_conn_id: str = "hive_cli_default",
+        hive_cli_conn_id: str = default_conn_name,
```

Comment:
We could, if we wanted, use `inspect.signature` to get the default value for the hook:

```ipython
In [3]: from airflow.providers.apache.hive.hooks.hive import HiveCliHook                                                                                                                                                                                                                                    

In [4]: import inspect                                                                                                                                                                                                                                                                                      

In [5]: inspect.signature(HiveCliHook).parameters['hive_cli_conn_id'].default                                                                                                                                                                                                                               
Out[5]: 'hive_cli_default'
```

Not needed, and might be overkill, but is just one less thing to need in the "interface"

Reply:
I want to do a miniumum set of changes for this discovery to pass, we can definitely improve it in the future.

Reply:
I really want to avoid creating new "Hook" interface now. My goal is to have provider hook discovery for now. Adding, standardizing documenting and polishing "common Hook API"  should be next step IMHO - possibly at the time when we introduce reading Hooks via " plugin-lilke" interface.

Reply:
You might not have explicitly stated it, but this is creating a new Hook interface.

I guess the main thing is this is opt-in, so existing sub-classes don't suddenly break. Good point. :+1: